pandas/tests/extension/base/ops.py

```diff
@@ -132,10 +132,8 @@ class BaseComparisonOpsTests(BaseOpsUtil):
     def _compare_other(self, s, data, op_name, other):
         op = self.get_op_from_name(op_name)
         if op_name == "__eq__":
-            assert getattr(data, op_name)(other) is NotImplemented
```

Comment:
We definitely don't want to remove this :) You'll need to revert it and whatever change required it :)

Reply:
@TomAugspurger I think this is actually correct to change this. As far as I understand, we only tested this as the base version (if you don't define an `__eq__` on your class, python will return this, I assume). But now we changed the base class to actually have a `__eq__` but that raise an AbstractError, so it needs to always be implemented, and thus can return something here that is not NotImplemented.

Reply:
What's `other` here? If it's a Series / DataFrame we need to make sure to return `NotImplemented` right? I thought that was the intent of this test, though I might misunderstand.

Reply:
`data` is the EA, and other can eg be a scalar. 

There is a separate test just below (that I added back) that explicitly tests that ``NotImplemented`` is returned for `EA.__eq__(Series/Index)`